### PR TITLE
Add lease resource name customization using flag

### PR DESCRIPTION
**Pull request by Duke0404**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Allows lease resource name used during leader election to be customized by passing a flag
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Enables ability to set custom lease resource name using flag
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster-autoscaler/main.go [hunk 1]
```diff
@@ -110,6 +110,7 @@ func multiStringFlag(name string, usage string) *MultiStringFlag {
 }
 
 var (
+	resourceName            = flag.String("lease-resource-name", "cluster-autoscaler", "The name of the cluster-autoscaler kubernetes resource for lease locking")
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/main.go:113]: Please change variable name to match the flag - `leaseResourceName`.

Nit regarding style: description could probably be simplified to sth like "The lease resource to use in leader election."

Comment (Duke0404) [reply cluster-autoscaler/main.go]: Updated accordingly

**Pull request merged**